garage/contrib/exp/experiment.py

```diff
@@ -0,0 +1,58 @@
+import gym
+
+from garage.contrib.exp import Agent
+from garage.contrib.exp import Checkpointer
+from garage.contrib.exp import Logger
+from garage.contrib.exp import Sampler
+
+
+class Experiment():
+    def __init__(
+            self,
+            env: gym.Env,
+            agent: Agent,
+        checkpointer: Checkpointer,
+            logger: Logger,
+        sampler: Sampler = None,
+        # common experiment parameters:
+            n_itr=10,
+            batch_size=1000,
+    ):
+        if sampler is None:
+            from garage.contrib.exp.samplers import BatchSampler
+            sampler = BatchSampler(env=env)
+
+        if checkpointer.resume:
```

Comment:
this makes `.resume` part of the public `Checkpointer` interface, but it's not in the ABC